airflow/configuration.py

```diff
@@ -1474,84 +1679,123 @@ def _generate_fernet_key() -> str:
-
-        log.info("Reading the config from %s", AIRFLOW_CONFIG)
+                airflow_config_parser.write(file)
+        make_group_other_inaccessible(AIRFLOW_CONFIG)
+    log.info("Reading the config from %s", AIRFLOW_CONFIG)
+    airflow_config_parser.read(AIRFLOW_CONFIG)
+    if airflow_config_parser.has_option("core", "AIRFLOW_HOME"):
+        msg = (
+            "Specifying both AIRFLOW_HOME environment variable and airflow_home "
+            "in the config file is deprecated. Please use only the AIRFLOW_HOME "
+            "environment variable and remove the config file entry."
+        )
+        if "AIRFLOW_HOME" in os.environ:
+            warnings.warn(msg, category=DeprecationWarning)
+        elif airflow_config_parser.get("core", "airflow_home") == AIRFLOW_HOME:
+            warnings.warn(
+                "Specifying airflow_home in the config file is deprecated. As you "
+                "have left it at the default value you should remove the setting "
+                "from your airflow.cfg and suffer no change in behaviour.",
+                category=DeprecationWarning,
+            )
+        else:
+            # there
+            AIRFLOW_HOME = airflow_config_parser.get("core", "airflow_home")  # type: ignore[assignment]
+            warnings.warn(msg, category=DeprecationWarning)
 
-        local_conf.read(AIRFLOW_CONFIG)
 
-        if local_conf.has_option("core", "AIRFLOW_HOME"):
-            msg = (
-                "Specifying both AIRFLOW_HOME environment variable and airflow_home "
-                "in the config file is deprecated. Please use only the AIRFLOW_HOME "
-                "environment variable and remove the config file entry."
-            )
-            if "AIRFLOW_HOME" in os.environ:
-                warnings.warn(msg, category=DeprecationWarning)
-            elif local_conf.get("core", "airflow_home") == AIRFLOW_HOME:
-                warnings.warn(
-                    "Specifying airflow_home in the config file is deprecated. As you "
-                    "have left it at the default value you should remove the setting "
-                    "from your airflow.cfg and suffer no change in behaviour.",
-                    category=DeprecationWarning,
-                )
-            else:
-                # there
-                AIRFLOW_HOME = local_conf.get("core", "airflow_home")  # type: ignore[assignment]
-                warnings.warn(msg, category=DeprecationWarning)
+def initialize_config() -> AirflowConfigParser:
+    """
+    Load the Airflow config files.
 
-        # They _might_ have set unit_test_mode in the airflow.cfg, we still
-        # want to respect that and then load the unittests.cfg
-        if local_conf.getboolean("core", "unit_test_mode"):
-            local_conf.load_test_config()
+    Called for you automatically as part of the Airflow boot process.
+    """
+    global FERNET_KEY, AIRFLOW_HOME, WEBSERVER_CONFIG
```

Comment:
```suggestion
    global WEBSERVER_CONFIG
```

Don't we just need this one? We aren't doing anything with the other 2?

Reply:
They are actually used by

```
def get_all_expansion_variables() -> dict[str, Any]:
    return {k: v for d in [globals(), locals()] for k, v in d.items()}
```